accepted/2020/minimum-os-version/minimum-os-version.md

```diff
@@ -133,48 +129,48 @@ if (RuntimeInformation.CheckOSVersion(15)) {
 }
 ```
 
-There will be an additional overload that checks the OS as well as the OS version, for use in portable libraries:
+There will be another method on `RuntimeInformation` that checks the OS as well as the OS version, for use in platform-neutral libraries that use runtime checks to support or light up features on multiple OSes:
 
 ```csharp
-public static bool RuntimeInformation.CheckOS(
+public static bool CheckOS(
     string platformIdentifier,
     int majorVersion, int minorVersion = 0, int servicingVersion = 0
 );
 ```
 
 As with the availability attributes, there could be an additional overload that used enums and target typing to improve readability.
 
-> NOTE: The linker, JIT and/or AOT compiler could treat these method as intrinsics with constant values, which would enable it to eliminate unnecessary fallback code when consuming portable libraries or libraries with a lower OS minimum version than the app itself.
+***NOTE****: The linker, JIT and/or AOT compiler could treat these method as intrinsics with constant values, which would enable it to eliminate unnecessary fallback code when consuming portable libraries or libraries with a lower OS minimum version than the app itself.*
 
 # MSBuild Properties
 
 The `Microsoft.Net.Sdk` targets will extract the OS API Version component from the `TargetFramework` into the `TargetPlatformVersion` MSBuild property and burn it into an assembly attribute. Tools that use this value are expected to access it from the MSBuild property or from the assembly attribute.
 
-Project files may specify a `MinimumPlatformVersion`, and if they do not it will default to the `TargetPlatformVersion` value. The `MinimumPlatformVersion` must not be higher than the `TargetPlatformVersion`, and this will be validated at the start of the build.
+Project files may specify an `OSMinimumVersion`, and if they do not it will default to OS Version equivalent to the OS API version specified in the `TargetPlatformVersion` value. The `OSMinimumVersion` must not be higher than the `TargetPlatformVersion`, and this will be validated at the start of the build.
```

Comment:
How will we handle this if we update the bindings outside of the OS version.  IE we ship iOS bindings 15.0.1, but it still maps to iOS 15.0.  Does each workload need to keep a mapping between `TargetPlatformVersion` and `OSMinimumVersion` for that platform?

Reply:
Yes, that's unfortunate but necessary. Still thinking about whether to formally define a standard mechanism or leave it up to the platform targets.

Reply:
Why not define a roll forward mechanism for bindings just like the runtime packages?

That would certainly remove these complications!!

Reply:
@Nirmal4G it's not that easy.

For example: from the TFM `net5-ios15.0` we can infer that the `OSMinimumVersion` default is `15.0`. However, suppose the TFM `net5-ios15.0.1` was a revision of the bindings rather than bindings for a new OS version, then its `OSMinimumVersion` would also be need to be `15.0`.

We could handle that in a general way by saying that only the first two components of the binding version are the OS version and the third component is the bindings revision. However, those kinds of conventions we need to be determined on a platform by platform basis, as different platforms have different versioning schemes.

Reply:
This is still a problem in the current proposal with the TF having bindings revision. Do we expect people to target the same platform for each bindings revision, I don't think so. I didn't encounter any packages targeting specifically any platform with multiple bindings!!! Are there?

If so, Why not separate the bindings version into a separate property?

Say `PlatformPackageVersion` if we're using nuget as the delivery mechanism for the bindings and `PlatformBindingsRevision` if we're shipping bindings within the workload itself. If we're using both we could use the latter to set the former for resolving the implicit FrameworkReference.